2324-wt/abstract-interpretation/interim/poster.tex

```diff
@@ -0,0 +1,221 @@
+	\columnbreak
+	\begin{tikzpicture}
+		\node[n,roundednode] (X1) {\rc{x}\textsubscript1};
+		\node[n,roundnode,right=of X1] (X2) {\rc{x}\textsubscript2};
+		\node[n,roundednode,right=of X2] (Y1) {\rc{y}\textsubscript1};
+		\node[n,roundnode,right=of Y1] (Y2) {\rc{y}\textsubscript3};
+		\node[n,squarenode,right=of Y2] (C1) {\rc{compA}};
+
+		\node[n,roundnode,below=of Y2] (Y3) {\rc{y}\textsubscript4};
+		\node[n,squarenode,below=of C1] (C2) {\rc{compB}};
+
+		\node[n,roundnode,above=of Y1] (Y4) {\rc{y}\textsubscript2};
+
+		\node[dr] at (X1.north east) {\tiny$\{10\}$};
+		\node[ds] at (X2.north east) {\tiny$\{10\}$};
+		\node[dr] at (Y1.north east) {\tiny$\{15\}$};
+		\node[ds] at (Y2.north east) {\tiny$\{15\}$};
+		\node[ds] at (Y3.north east) {\tiny\tbot};
+		\node[ds] at (Y4.north east) {\tiny$\{15\}$};
+
+		\node[t,fit=(X1)(X2)(Y1)(Y2)(Y3)(Y4)(C1)(C2)] (F) {};
+		% \node[label] at (F.south) {Step 2};
+
+		\draw[a] (X1) to (X2);
+		\draw[a] (X2) to (Y1);
+		\draw[a] (Y1) to (Y2);
+		\draw[a] (Y2) to (C1);
+
+		\draw[a] (Y1) to (Y3);
+		\draw[a] (Y3) to (C2);
+
+		\draw[a] (Y1) to (Y4);
+	\end{tikzpicture}\par
+	\paragraph{Step 2} In the second step happens the actual abstract interpretation.
+	This step will gradually decorate the dataflow graph with the domains of the
+	respective variables.\par
+	\columnbreak
+	\begin{tikzpicture}
+		\node[n,roundednode] (X1) {\rc{x}\textsubscript1};
+		\node[n,roundnode,right=of X1] (X2) {\rc{x}\textsubscript2};
+		\node[n,roundednode,right=of X2] (Y1) {\rc{y}\textsubscript1};
+		\node[n,roundnode,right=of Y1] (Y2) {\rc{y}\textsubscript3};
+		\node[n,squarenode,right=of Y2] (C1) {\rc{compA}};
+
+		\begin{scope}[no,transparency group]
+			\node[n,roundnode,below=of Y2] (Y3) {\rc{y}\textsubscript4};
+			\node[n,squarenode,below=of C1] (C2) {\rc{compB}};
+			\node[ds] at (Y3.north east) {\tiny\tbot};
+		\end{scope}
+
+		\node[n,roundnode,above=of Y1] (Y4) {\rc{y}\textsubscript2};
+
+		\node[dr] at (X1.north east) {\tiny$\{10\}$};
+		\node[ds] at (X2.north east) {\tiny$\{10\}$};
+		\node[dr] at (Y1.north east) {\tiny$\{15\}$};
+		\node[ds] at (Y2.north east) {\tiny$\{15\}$};
+		\node[ds] at (Y4.north east) {\tiny$\{15\}$};
+
+		\node[t,fit=(X1)(X2)(Y1)(Y2)(Y3)(Y4)(C1)(C2)] (F) {};
+		% \node[label] at (F.south) {Step 3};
+
+		\draw[a] (X1) to (X2);
+		\draw[a] (X2) to (Y1);
+		\draw[a] (Y1) to (Y2);
+		\draw[a] (Y2) to (C1);
+
+		\begin{scope}[no,transparency group]
+			\draw[a] (Y1) to (Y3);
+			\draw[a] (Y3) to (C2);
+		\end{scope}
+
+		\draw[a] (Y1) to (Y4);
+	\end{tikzpicture}\par
+	% \paragraph{Step 3} The last step uses the decorated dataflow graph to determine wich
+	% path can never be executed. The identified path will then be discarded from the
+	% graph.\par
+	\paragraph{Step 3} The last step uses the decorated dataflow graph to determine and
+	track different properties. We can, e.g., catch endless loops, or, as shown above,
+	eliminate dead paths in the dataflow graph.\par
```

Comment:
dead paths suggeriert wieder nen CFG - vlt "ignore dead code"?

Reply:
Fixed in ad36492